ocaml/xapi/cert_distrib.ml

```diff
@@ -372,21 +372,77 @@ let exchange_certificates_among_all_members ~__context =
          we do not guarantee 'atomicity', so if regenerating the bundle on one host
          fails, then state across the pool will most likely become inconsistent, and
          manual intervention may be required *)
+  let maybe_insert_fist =
+    (* if there is a fist point:
+     *   - throw an error at a random point
+     *   - print out what is going to execute for debugging purposes
+     *)
+    match Xapi_fist.exchange_certificates_in_pool () with
+    | None ->
+        Fun.id
+    | Some seed ->
+        fun ops ->
+          Random.init seed ;
+          let rand_i = Random.int (List.length ops) in
+          let throw_op =
+            ( "FIST"
+            , fun () ->
+                raise
+                  Api_errors.(
+                    Server_error
+                      ( internal_error
+                      , ["/tmp/fist_exchange_certificates_in_pool FIST!"]
+                      )
+                  )
+            )
+          in
+          let ops' =
+            List.fold_left
```

Comment:
Fold_left tends to make transformation on lists harder to inderstand. This looks like it has the same behaviour as `Xapi_stdext_std.Listext.List.take (rand_i - 1) @ [throw_op]`, this is not very efficient, adapting the code for `take` might be best here:
https://github.com/xapi-project/stdext/blob/master/lib/xapi-stdext-std/listext.ml#L61
```
  let take_and_append n op list =
    let rec loop i acc = function
      | x :: xs when (i - 1) < n ->
          loop (i + 1) (x :: acc) xs
      | _ ->
          List.rev acc
    in
    loop 0 [op] list
```